compiler/il/symbol/OMRResolvedMethodSymbol.cpp

```diff
@@ -628,7 +629,12 @@ OMR::ResolvedMethodSymbol::induceOSRAfter(TR::TreeTop *insertionPoint, TR_ByteCo
       osrBlock->getEntry()->getNode()->setByteCodeInfo(induceBCI);
       osrBlock->getExit()->getNode()->setByteCodeInfo(induceBCI);
 
-      cfg->findLastTreeTop()->join(osrBlock->getEntry());
+      // Load the cached CFG end if available, and store it when finished
```

Comment:
How do we ensure that no other logic (unrelated to adding OSR induction logic) appends code at the end of the list of treetops making the cached value "stale" ? 

I also find the reference to the "CFG" in this change a bit clunky in that what is being talked about is the end of the treetops in list order, which would seem to be unrelated to the CFG directly as such. The "CFG end" is already a phrase that has a meaning in the compiler, namely the dummy block without any treetops associated with it that all blocks ending in a return have CFG edge(s) to.

Reply:
Other than OSR induction logic, no other blocks should be added by `induceOSRAfter `. If the caller could potentially add blocks between calls to this method, they must update the cached treetop accordingly. An assert has been added to reflect the final treetop must have no trees following it.

I have corrected the names to indicate the last treetop rather than the CFG end.